hack/run-in-python-container.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs $@ in a python container
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+cd "${REPO_ROOT}"
+
+PY_IMAGE='python:3.9-slim-buster'
+PYLINTHOME="$(mktemp -d)"
+
+# Technically we can assume that this is always set, setting here again to
+# ensure consistency
+VIRTUAL_ENV="${VIRTUAL_ENV:-${REPO_ROOT:?}/.python_virtual_env}"
+
+# Setting PATH is a trick learnt from
+# https://pythonspeed.com/articles/activate-virtualenv-dockerfile/, for
+# mimicking the behavior of `source venv_dir/bin/activate`, same as setting
+# PYTHONPATH to empty. Tried several other alternatives:
+#   - Set PATH in docker image. This doesn't work because the workspace is set
+#     afterwards.
+#   - Run `source venv_dir/bin/activate` as part of docker run argument. This
+#     cannot be done directly as `source` is a bash builtin, so have to do `bash
+#     -c `source.., python3..` stuff, and the quotes passing around almost
```

Comment:
I think just `docker run ... "${PY_IMAGE}" bash -c 'source venv_dir/bin/activate && exec "$@"' " $@"` should do what you describe here.

Reply:
I'll give it a shot

Reply:
good to learn this trick, seems like it should work, until another day of learning again "bash is hard"! I got stuck trying to understand this:

```
<script.sh>
echo "outside: $@"
bash -c 'echo "inside: $@"' "$@"
```

```
$ script.sh pip3 install -r requirements3.txt

outside: pip3 install -r requirements3.txt
inside: install -r requirements3.txt
```

somehow `pip3` disappeared

Reply:
Ah right, it will need to include `"$0 $@"` in the inner exec, because $0 is dropped from $@ each time.

The first time this is desired here. Repeat times it is not.

`argv[0]` should contain the pathname of the executable being invoked, so programs like bash treat it specially and it is not in $@ as a program argument (which is ~`argv[1:]`).

Reference for more: https://man7.org/linux/man-pages/man2/execve.2.html

Reply:
had an offline chat with Ben, though this works, the nested quotes don't look elegant. Tried to introduce `activate.sh` script that is depended by other python test scripts, which does the activation, seems like a good solution, only down side is more scripts